processes/ingest/doab.py

```diff
@@ -34,35 +41,31 @@ def __init__(self, *args):
 
     def runProcess(self):
         try:
-            self.generateEngine()
-            self.createSession()
-
             records = []
 
             if self.process == 'daily':
                 records = self.dspace_service.get_records(offset=self.offset, limit=self.limit)
             elif self.process == 'complete':
                 records = self.dspace_service.get_records(full_import=True, offset=self.offset, limit=self.limit)
             elif self.process == 'custom':
-                records = self.dspace_service.get_records(start_timestamp=self.ingestPeriod, offset=self.offset, limit=self.limit)
-            elif self.singleRecord:
-                record = self.dspace_service.get_single_record(record_id=self.singleRecord, source_identifier=self.DOAB_IDENTIFIER)
+                records = self.dspace_service.get_records(start_timestamp=self.ingest_period, offset=self.offset, limit=self.limit)
+            elif self.single_record:
+                record = self.dspace_service.get_single_record(record_id=self.single_record, source_identifier=self.DOAB_IDENTIFIER)
                 self.manage_links(record)
+                self.record_buffer.add(record.record)
 
             if records:
                 for record in records:
                     self.manage_links(record)
+                    self.record_buffer.add(record.record)
+            
+            self.record_buffer.flush()
 
-            self.saveRecords()
-            self.commitChanges()
-
-            logger.info(f'Ingested {len(self.records) if records else 1} DOAB records')
+            logger.info(f'Ingested {self.record_buffer.ingest_count} DOAB records')
 
         except Exception as e:
             logger.exception('Failed to run DOAB process')
             raise e
-        finally:
-            self.close_connection()
```

Comment:
I think we should keep this close_connection method to ensure the session is ended after the flush. What do you think?

Reply:
Good idea, re-added it